openedx/core/djangoapps/programs/utils.py

```diff
@@ -383,27 +384,33 @@ def _attach_run_mode_course_url(self, run_mode):
         run_mode['course_url'] = reverse('course_root', args=[self.course_key])
 
     def _attach_run_mode_end_date(self, run_mode):
-        run_mode['end_date'] = self.course_overview.end_datetime_text()
+        run_mode['end_date'] = self.course_overview.end
 
     def _attach_run_mode_enrollment_open_date(self, run_mode):
         run_mode['enrollment_open_date'] = strftime_localized(self.enrollment_start, 'SHORT_DATE')
 
     def _attach_run_mode_is_course_ended(self, run_mode):
-        end_date = self.course_overview.end or datetime.datetime.max.replace(tzinfo=pytz.UTC)
-        run_mode['is_course_ended'] = end_date < timezone.now()
+        end_date = self.course_overview.end or datetime.datetime.max.replace(tzinfo=utc)
+        run_mode['is_course_ended'] = end_date < datetime.datetime.now(utc)
 
     def _attach_run_mode_is_enrolled(self, run_mode):
         run_mode['is_enrolled'] = CourseEnrollment.is_enrolled(self.user, self.course_key)
 
+    def _attach_run_mode_user_preferences(self, run_mode):
+        run_mode['user_preferences'] = get_user_preferences(self.user)
+
     def _attach_run_mode_is_enrollment_open(self, run_mode):
-        enrollment_end = self.course_overview.enrollment_end or datetime.datetime.max.replace(tzinfo=pytz.UTC)
-        run_mode['is_enrollment_open'] = self.enrollment_start <= timezone.now() < enrollment_end
+        enrollment_end = self.course_overview.enrollment_end or datetime.datetime.max.replace(tzinfo=utc)
+        run_mode['is_enrollment_open'] = self.enrollment_start <= datetime.datetime.now(utc) < enrollment_end
 
     def _attach_run_mode_marketing_url(self, run_mode):
         run_mode['marketing_url'] = get_run_marketing_url(self.course_key, self.user)
 
     def _attach_run_mode_start_date(self, run_mode):
-        run_mode['start_date'] = self.course_overview.start_datetime_text()
+        run_mode['start_date'] = self.course_overview.start
+
+    def _attach_run_mode_advertised_start(self, run_mode):
+        run_mode['advertised_start'] = self.course_overview.advertised_start
```

Comment:
Is your concern here that the `start_date` may not always be the `advertised_start`, and that the `advertised_start` should be displayed if available? I'm not sure I would introduce this now. It seems beyond the scope of your work here. We'll also be switching this to use data from the catalog service soon (which takes care of providing correct start dates).